### PR TITLE
[core] Replace `classnames` utility with `clsx` dependency

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Part of #6 : To align convention with the main core repository. Also faster execution - [clsx performance benchmarks](https://github.com/lukeed/clsx/tree/master/bench) 
Part of #18 : Bundle size reduction. Not sure if `classnames` custom utility is smaller or `clsx` dependency.

Looked worthwhile to do this now and as long as the codebase is small it would be less cumbersome to replace now than doing it later when codebase grows. 

Note: We already also use the [babel-plugin-optimize-clsx](https://github.com/merceyz/babel-plugin-optimize-clsx) 
https://github.com/JedWatson/classnames/issues/187#issuecomment-504311656

### CODE REVIEW

**Comment by oliviertassinari** [comment]
> so it's easy to switch the underlying implementation.

Why is making it easy to switch something to optimize for? I mean it seems that it was simple enough in this very occurrence. Could it make it more confusing for developers when contributing between different components? TreeView, Scheduler, etc.

**Comment by dtassone** [comment]
> > so it's easy to switch the underlying implementation.
> 
> Why is making it easy to switch something to optimize for? I mean it seems that it was simple enough in this very occurrence. Could it make it more confusing for developers when contributing between different components? TreeView, Scheduler, etc.

Bundle size reduction. Not sure if classnames custom utility is smaller or clsx dependency. => Just to be able to easily switch component by adding an abstraction that could point here or there. 

In the past, we have changed the debounce function several times.

**Comment by oliviertassinari** [comment]
> Not sure if classnames custom utility is smaller or clsx dependency.

@dtassone the data grid imports *clsx* by importing the core components. So *classnames + clsx* (HEAD) will always be larger than *clsx* only (PR), hence why the bundle size win of this PR. Also, a good reminder to work on #37's item bundle size diff, so this can be seen right from the PR review with an automatic report.

> to be able to easily switch component by adding an abstraction that could point here or there.

@ZeeshanTamboli Was doing the change in this PR, easy enough?

File: packages/grid/data-grid/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "@material-ui/utils": "^5.0.0-alpha.14",
+    "clsx": "^1.1.1",
```
Comment (oliviertassinari) [inline packages/grid/data-grid/package.json:34]: A small detail, 

```suggestion
    "clsx": "^1.0.4",
```

x2 might be better https://github.com/mui-org/material-ui/blob/8cb115b4fa8bb9ba0690b50bae1c1ffbbb7346d1/packages/material-ui/package.json#L72

**Comment by ZeeshanTamboli** [comment]
> @ZeeshanTamboli Was doing the change in this PR, easy enough?

@oliviertassinari Yes.

File: packages/grid/data-grid/package.json (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/data-grid/package.json]: It's not really important

**Review by oliviertassinari — APPROVED** [review]

**Comment by dtassone** [comment]
> > Not sure if classnames custom utility is smaller or clsx dependency.
> 
> @dtassone the data grid imports _clsx_ by importing the core components. So _classnames + clsx_ (HEAD) will always be larger than _clsx_ only (PR), hence why the bundle size win of this PR. Also, a good reminder to work on #37's item "bundle size diff", so we can see the actual win right from the PR review with an automatic report.
> 
> > to be able to easily switch component by adding an abstraction that could point here or there.
> 
> @ZeeshanTamboli Was doing the change in this PR, easy enough?

Classname.ts
```tsx
import clsx from 'classnames'
export classnames= clsx
```
So if you want to unplug clsx, it is only in one file.
let say you want to replace it with material-ui/utils, you just have to change 1 file. 
That's what I meant.

**Pull request merged**